Library/Formula/spidermonkey.rb

```diff
@@ -1,18 +1,19 @@
 require 'formula'
 
 class Spidermonkey < Formula
-  homepage 'https://developer.mozilla.org/en/SpiderMonkey'
-  url 'http://ftp.mozilla.org/pub/mozilla.org/js/js185-1.0.0.tar.gz'
-  version '1.8.5'
-  sha1 '52a01449c48d7a117b35f213d3e4263578d846d6'
-
-  head 'https://hg.mozilla.org/tracemonkey/archive/tip.tar.gz'
-
-  bottle do
-    revision 1
-    sha1 "6c6fd6d40d41764a086a6fb134176253deb1a51b" => :yosemite
-    sha1 "5d19010b10a5f1827511ca791debf9f2d9076e47" => :mavericks
-    sha1 "37d04b64aba47dbf65f197aec94da9acf5f1fd4c" => :mountain_lion
+  homepage 'https://developer.mozilla.org/en-US/docs/Mozilla/Projects/SpiderMonkey'
+  url 'https://ftp.mozilla.org/pub/mozilla.org/js/mozjs-24.2.0.tar.bz2'
+  version '24.2.0'
+  sha1 'ce779081cc11bd0c871c6f303fc4a0091cf4fe66'
+
+  depends_on 'yasm'
+  depends_on 'gawk'
+  depends_on 'libidl'
+  depends_on 'ccache'
+
+  head do
+    depends_on 'https://raw.github.com/Homebrew/homebrew-versions/master/autoconf213.rb'
+    url 'https://hg.mozilla.org/mozilla-central/archive/tip.tar.bz2'
```

Comment:
well, on my gigabit internet connection, downloading `tip.tar.gz` takes about 6min at 503kB/s, whereas checking out the mercurial repository takes about 18min, making it about 170kB/s (because even though the downloading should be as fast, as it's originated from the same server but applying 200k patches in mercurial takes some time). Another thing is that the full repository for mozilla is about 3GB, whereas the tip archive is only 170MB and contains all that is needed for compilation.

And your comment would be relevant if brew was keeping the source checkout after installation, but as this is all done in a temporary directory cache, if you want to upgrade your tools, you'll actually checkout *again* the whole 3GB through mercurial, taking another 20 min.

Finally, as not everybody has a gigabit connection or >3GB available for compilation (how many 128GB laptops are there around, because of the SSD-based macs?), I believe it's a way more sensible choice to use the tip archive than the repository.

As a final thought, I believe that the main problem is that the mozilla people are using mercurial like svn or cvs, having all project in one huge repository, making it hard to extract the project of interest from the whole mess. It would be a smart move if they were creating a repository per project, and have the mozilla-central repository have them all as submodules.